pytaser/generator.py

```diff
@@ -557,116 +223,50 @@ def generate_tas(self, energy_min, energy_max, gaussian_width=0.2, step=0.1):
+                            self.occs_light[spin],
                             energy_mesh_ev,
-                            self.kpoint_index,
                             self.kpoint_weights,
                             gaussian_width,
+                            spin=spin,
                         )
                         jd_dark = jdos(
                             self.bs,
                             f,
                             i,
-                            self.occs_dark,
+                            self.occs_dark[spin],
                             energy_mesh_ev,
-                            self.kpoint_index,
                             self.kpoint_weights,
                             gaussian_width,
+                            spin=spin,
                         )
                         tas = jd_light - jd_dark
-                        jdos_dark_if[(i, f)] = jd_dark
-                        jdos_light_if[(i, f)] = jd_light
                         jdos_dark_cumulative += jd_dark
                         jdos_light_cumulative += jd_light
-                        tas_if[(i, f)] = tas
                         tas_cumulative += tas
-                        # denom_TAS_if[(i,f, "up")] = tas/jd_light
-                        # denom_TAS_cumulative = tas/jd_light
 
-        elif self.sppol == True:
-            for i in self.band_index:
-                for f in self.band_index:
-                    if f > i:
-                        jd_light = jdos(
-                            self.bs,
-                            f,
-                            i,
-                            self.occs_light,
-                            energy_mesh_ev,
-                            self.kpoint_index,
-                            self.kpoint_weights,
-                            gaussian_width,
-                        )
-                        jd_dark = jdos(
-                            self.bs,
-                            f,
-                            i,
-                            self.occs_dark,
-                            energy_mesh_ev,
-                            self.kpoint_index,
-                            self.kpoint_weights,
-                            gaussian_width,
-                        )
-                        tas = jd_light - jd_dark
-                        jdos_dark_if[(i, f, "up")] = jd_dark
-                        jdos_light_if[(i, f, "up")] = jd_light
-                        jdos_dark_cumulative += jd_dark
-                        jdos_light_cumulative += jd_light
-                        tas_if[(i, f, "up")] = tas
-                        tas_cumulative += tas
-                        # denom_TAS_if[(i,f, "up")] = tas/jd_light
-                        # denom_TAS_cumulative = tas/jd_light
+                        if self.bs.is_spin_polarized:
+                            spin_str = "up" if spin == Spin.up else "down"
+                            key = (i, f, spin_str)
```

Comment:
If the system is spin polarised this will keep the correct labelling of the bands.